dht.go

```diff
@@ -528,42 +522,63 @@ func (dht *DHT) get(key Hash, statusMask int, getMask int) (data []byte, entryTy
 	return
 }
 
-// _putLink is a low level routine to add a link, also used by mod
-func _putLink(tx *buntdb.Tx, base string, link string, tag string, src peer.ID) (err error) {
+// _link is a low level routine to add a link, also used by delLink
+// this ensure monotonic recording of linking attempts
+func _link(tx *buntdb.Tx, base string, link string, tag string, src peer.ID, status int, linkingEntryHash Hash) (err error) {
 	key := "link:" + base + ":" + link + ":" + tag
-	var val string
-	val, err = tx.Get(key)
-	if err == buntdb.ErrNotFound {
-		entryString, _ := json.Marshal(LinkEntry{StatusLive, peer.IDB58Encode(src)})
-		_, _, err = tx.Set(key, string(entryString), nil)
-		if err != nil {
+	_, err = tx.Get(key)
+	source := peer.IDB58Encode(src)
+	lehStr := linkingEntryHash.String()
+	var records []LinkEvent
+	if err == nil {
+		// TODO: if the link exists, then load the statuses and see
+		// what we should do about this situation
+		/*
```

Comment:
So it is ok to just ignore this case?!

Reply:
hm.. but shouldn't you just at least get the value and unmarshal the array so you can really append something below (which is maybe why you have an append there which right now operates always on an empty slice)?